types/schema/File.ts

```diff
@@ -26,3 +26,14 @@ type FileTypeMap = {
  * @description Types of planning documents and drawings
  */
 export type FileType = FileTypeMap[keyof FileTypeMap];
+
+/**
+ * @id #FileCondition
+ */
+export type FileCondition = 'required' | 'recommended' | 'optional';
+
+/**
+ * @id #RequestedFileType
+ * @description Types of planning documents and drawings, with an attached condition as presented to the applicant
+ */
+export type RequestedFileType = FileType & {condition: FileCondition};
```

Comment:
As we're relying on the enum `FileTypes` to dynamically generate the `FileType` type, we can't extend this interface - 

![image](https://github.com/theopensystemslab/digital-planning-data-schemas/assets/20502206/2d08abe7-6e9b-42a9-beab-467d58547f83)

This means that the schema repeats the file list (have a look at the diff in `schema.json`).

I spent a fair bit of time trying to work around this to no success. The closest would be something like - 

```ts
export type RequestedFileType = {file: FileType} & {condition: FileCondition};
```

Both have some compromises for the consumer. I'll reach out to the BOPS team to explain this.

I think we're fine to proceed as-is, we can maybe re-evaluate how we handle these enum types at some point (see https://github.com/theopensystemslab/digital-planning-data-schemas/issues/51)